### ISSUE TITLE
module_test_ios fails, Undefined symbols for architecture armv7: "_OBJC_CLASS_$_UIPointerShape", referenced from: objc-class-ref in GoogleMaps

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#63447]
```
[module_test_ios] [STDOUT] ══════════════╡ ••• Build ephemeral host app with CocoaPods ••• ╞═══════════════
[module_test_ios] [STDOUT] 
[module_test_ios] [STDOUT] 
[module_test_ios] [STDOUT] Executing: /private/tmp/flutter sdk/bin/flutter build ios --no-codesign in /var/folders/3y/l0z1x3693dl_8n0qybp4dqwh0000gn/T/flutter_module_test.lFLCqR/hello
[module_test_ios] [STDOUT] stdout: The plugin `flutter_plugin_android_lifecycle` is built using an older version of the Android plugin API which assumes that it's running in a full-Flutter environment. It may have undefined behaviors when Flutter is integrated into an existing app as a module.
[module_test_ios] [STDOUT] stdout: The plugin can be updated to the v2 Android Plugin APIs by following https://flutter.dev/go/android-plugin-migration.
[module_test_ios] [STDOUT] stdout: Warning: Building for device with codesigning disabled. You will have to manually codesign before deploying to device.
[module_test_ios] [STDOUT] stdout: Building io.flutter.devicelab.hello for device (ios-release)...
[module_test_ios] [STDOUT] stdout: Running pod install...                                              5.6s
[module_test_ios] [STDOUT] stdout: Running Xcode build...                                          
[module_test_ios] [STDOUT] stdout: Xcode build done.                                           35.5s
[module_test_ios] [STDOUT] stdout: Failed to build iOS app
[module_test_ios] [STDOUT] stdout: Error output from Xcode build:
[module_test_ios] [STDOUT] stdout: ↳
[module_test_ios] [STDOUT] stdout:     ** BUILD FAILED **
[module_test_ios] [STDOUT] stdout: 
[module_test_ios] [STDOUT] stdout: 
[module_test_ios] [STDOUT] stdout: Xcode's output:
[module_test_ios] [STDOUT] stdout: ↳
[module_test_ios] [STDOUT] stdout:     Undefined symbols for architecture armv7:
[module_test_ios] [STDOUT] stdout:       "_OBJC_CLASS_$_UIPointerShape", referenced from:
[module_test_ios] [STDOUT] stdout:           objc-class-ref in GoogleMaps
[module_test_ios] [STDOUT] stdout:       "_OBJC_CLASS_$_UIPointerStyle", referenced from:
[module_test_ios] [STDOUT] stdout:           objc-class-ref in GoogleMaps
[module_test_ios] [STDOUT] stdout:     ld: symbol(s) not found for architecture armv7
[module_test_ios] [STDOUT] stdout:     clang: error: linker command failed with exit code 1 (use -v to see invocation)
[module_test_ios] [STDOUT] stdout:     note: Using new build system
[module_test_ios] [STDOUT] stdout:     note: Planning build
[module_test_ios] [STDOUT] stdout:     note: Constructing build description
```

**Comment by jmagman** [comment]
GoogleMaps 3.10.0 podspec was published at the same time.
https://github.com/CocoaPods/Specs/blob/master/Specs/a/d/d/GoogleMaps/3.10.0/GoogleMaps.podspec.json

**Comment by jmagman** [comment]
I'm actually not sure how this happens.  google_maps_flutter pub version is at 0.5.29 and GoogleMaps CocoaPods is at 3.10.0:
https://cirrus-ci.com/task/5509800735801344?command=main#L2220
```
[module_test_ios] [STDOUT]   google_maps_flutter:
[module_test_ios] [STDOUT]     dependency: "direct main"
[module_test_ios] [STDOUT]     description:
[module_test_ios] [STDOUT]       name: google_maps_flutter
[module_test_ios] [STDOUT]       url: "https://pub.dartlang.org"
[module_test_ios] [STDOUT]     source: hosted
[module_test_ios] [STDOUT]     version: "0.5.29"
```
```
[module_test_ios] [STDOUT]   - GoogleMaps (3.10.0):
[module_test_ios] [STDOUT]     - GoogleMaps/Maps (= 3.10.0)
```

However locally when I try to force this combination I get this error, which the test doesn't show:
```
[!] CocoaPods could not find compatible versions for pod "GoogleMaps":
  In Podfile:
    google_maps_flutter (from `.symlinks/plugins/google_maps_flutter/ios`) was resolved to 0.0.1, which depends on
      GoogleMaps (= 3.10.0)

Specs satisfying the `GoogleMaps (= 3.10.0)` dependency were found, but they required a higher minimum deployment target.
```
And that makes sense, because [google_maps_flutter minimum version is iOS 8.0](https://github.com/flutter/plugins/blob/7539aaed7fe7bf7fbedca3ec26359e40c9d80b73/packages/google_maps_flutter/google_maps_flutter/ios/google_maps_flutter.podspec#L22) and GoogleMaps 3.10 is iOS 9.0.
[GoogleMaps 2.7.0](https://github.com/CocoaPods/Specs/blob/master/Specs/a/d/d/GoogleMaps/2.7.0/GoogleMaps.podspec.json#L14) is the last version to have a minimum of 8.0:.

I _think_ the right thing to do it change the podspec to ` s.dependency 'GoogleMaps', '~> 2.7'` until that minimum version is increased to 9.0.

Not sure why this started happening as soon as GoogleMaps 3.10.0 was published.

\cc @ditman @cyanglaz

**Comment by ditman** [comment]
I have never seen this error before, I'm not touching podspecs (only the fake ones for web components). Odd.

> I think the right thing to do it change the podspec to s.dependency 'GoogleMaps', '~> 2.7' until that minimum version is increased to 9.0.

I'm trying to find other *similar* builds to see which version of GoogleMaps they're grabbing.

**Comment by ditman** [comment]
I think there are similarly-failing checks in flutter/plugins, with the Camera plugin:

https://cirrus-ci.com/task/5423395602956288

```
    ** BUILD FAILED **
Xcode's output:
↳
    /var/folders/3y/l0z1x3693dl_8n0qybp4dqwh0000gn/T/cirrus-ci-build/packages/camera/ios/Classes/CameraPlugin.m:66:24: warning: 'AVCapturePhotoOutput' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - (void)captureOutput:(AVCapturePhotoOutput *)output
                           ^
    In module 'AVFoundation' imported from /var/folders/3y/l0z1x3693dl_8n0qybp4dqwh0000gn/T/cirrus-ci-build/packages/camera/ios/Classes/CameraPlugin.m:6:
    /Applications/Xcode-11.3.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.2.sdk/System/Library/Frameworks/AVFoundation.framework/Headers/AVCapturePhotoOutput.h:48:12: note: 'AVCapturePhotoOutput' has been marked as being introduced in iOS 10.0 here, but the deployment target is iOS 8.0.0
```

Have iOS 8 and 9 died suddenly? :/

**Comment by jmagman** [comment]
@ditman That's warning is an old issue https://github.com/flutter/flutter/issues/20708.
Why is camera trying to call GoogleMaps code?
```
    CompileC /Users/anka/Library/Developer/Xcode/DerivedData/Runner-bxdmgdiknppixagjxwwbajlmfdkc/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/camera.build/Objects-normal/arm64/CameraPlugin.o /var/folders/3y/l0z1x3693dl_8n0qybp4dqwh0000gn/T/cirrus-ci-build/packages/camera/ios/Classes/CameraPlugin.m normal arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
    Undefined symbols for architecture armv7:
      "_OBJC_CLASS_$_UIPointerShape", referenced from:
          objc-class-ref in GoogleMaps
      "_OBJC_CLASS_$_UIPointerStyle", referenced from:
          objc-class-ref in GoogleMaps
```

**Comment by ditman** [comment]
@jmagman there's a couple of internal issues where this is reported: 

* b/163359804 
* b/163474612 

Seems to be a problem with v3.10, requiring a higher XCode than the one we're running our tests on (11.3.1 IIRC)

**Comment by jmagman** [comment]
That framework need to fix some availability, probably.  But the plugin can't use 3.10 anyway because it requires a minimum of iOS 9.0, and the plugin declares it supports 8.0.  The dependency should be updated to `~> 2.7`.

**Comment by jmagman** [comment]
Oh wait, https://github.com/flutter/plugins/pull/2920 was closed. Hmmm...

**Comment by ditman** [comment]
I'm trying to pin to 3.9, which is probably the last good version we were using anyways. Opening a new PR right now and seeing what happens.

**Comment by ditman** [comment]
Summarizing the "investigation" we've done so far:

* The minimum version of ios was bumped from 8.0 to 9.0 [recently](https://github.com/flutter/flutter/pull/62902).

* Turns out the `build_all_plugins_ipa` uses `flutter create` to [auto-generate an app](https://github.com/flutter/plugin_tools/blob/master/lib/src/create_all_plugins_app_command.dart#L52-L62) with all the plugins. Other tests may do the same.

* The google_maps_flutter plugin specifies a *minimum* version of ios to run (8.0). Resolving that in an app for 8.0, will yield GoogleMaps (pod) 2.7 (as seen [here](https://github.com/flutter/plugins/pull/2924#issuecomment-672290890)), however new apps that use 9.0 can resolve 3.x versions of the pod.

* 3.9 was good, as seen in the full-green pass linked in the comment above.

* 3.10 is bad, as linked in the buganizer issues in the PR; Probably because it exposes new APIs that require a newer version of XCode. The change is [undocumented](https://developers.google.com/maps/documentation/ios-sdk/releases) as of now, (_and this whole line is pure speculation_):

> Version 3.10.0 - August 10, 2020
>
> * Version bump, no changes. This version is functionally identical to v.3.9.0.
> * There is a new Maps Swift Demo App in Google Maps cocoapods.

For now, google_maps_flutter will pin GoogleMaps (pod) to `< 3.10`, that should be removed once the issue is fixed in the pod.

**Comment by ditman** [comment]
Cleanup ticket created: https://github.com/flutter/flutter/issues/63530

**Comment by ditman** [comment]
Fix to google_maps_flutter plugin tagged and published: [v0.5.29+1](https://pub.dev/packages/google_maps_flutter/changelog#05291)
